partiql-ast/src/ast.rs

```diff
@@ -394,7 +397,10 @@ pub enum CollectionLit {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum DateTimeLit {
     DateLit(String),
-    TimeLit(String),
+    TimeLit {
+        value: Option<String>,
```

Comment:
Why is the `value` string `Option`? Shouldn't it be a required field for `TimeLit`?

Reply:
Oh, is it because it's being reused as a type for `CAST` expressions?

Reply:
> Oh, is it because it's being reused as a type for `CAST` expressions?

Yes, or in `IS TIME`;